kanidm_unix_int/src/daemon.rs

```diff
@@ -393,29 +444,24 @@ async fn main() {
             debug!("Profile -> {}", env!("KANIDM_PROFILE_NAME"));
             debug!("CPU Flags -> {}", env!("KANIDM_CPU_FLAGS"));
 
-            let cfg_path = Path::new("/etc/kanidm/config");
-            let cfg_path_str = match cfg_path.to_str() {
-                Some(cps) => cps,
-                None => {
-                    error!("Unable to turn cfg_path to str");
-                    std::process::exit(1);
-                }
-            };
+
+            #[allow(clippy::expect_used)]
+            let cfg_path_str = clap_args.get_one::<String>("client-config").expect("Failed to pull the client config path");
+            let cfg_path: PathBuf =  PathBuf::from(cfg_path_str);
+
             if !cfg_path.exists() {
                 // there's no point trying to start up if we can't read a usable config!
                 error!(
                     "Client config missing from {} - cannot start up. Quitting.",
                     cfg_path_str
                 );
-                std::process::exit(1);
-            }
-
-            if cfg_path.exists() {
+                return
+            } else {
                 let cfg_meta = match metadata(&cfg_path) {
                     Ok(v) => v,
                     Err(e) => {
                         error!("Unable to read metadata for {} - {:?}", cfg_path_str, e);
-                        std::process::exit(1);
+                        return ExitCode::from(1)
```

Comment:
Did you rebuilt/test? I think the return has to have every return path as exitcode and the main function signature needs to change too? my guess is this won't build as is.

Reply:
ah poop I didn't mean to include this in the most recent push >.>